### ISSUE TITLE
adev: IP blocked by firewall

### ISSUE THREAD

**Issue by kienhoefr** [angular/angular#59062]
### Describe the problem that you experienced

Hi everyone,

I'm not sure if GitHub is the right place for this report, but I figured this is the best way to reach the people who know who to talk to.

Ever since the launch of angular.dev as the main website (since v18), we can no longer reliably access the site from our corporate network.
IT claims the (apparently shared) Google Cloud IPs used to serve the angular websites (.dev and .io) have bad reputation and are classified as botnet by our firewall vendor (Clavister).

Here are some public IP reputation listings that classify the IP as malicious:
- https://www.abuseipdb.com/check/199.36.158.100
- https://www.ipqualityscore.com/bot-management/bot-detection-check/lookup/199.36.158.100
- https://www.apivoid.com/tools/ip-reputation-check/

Therefore accessing any websites served over those IPs are blocked (inbound and outbound).
Sometimes accessing the site works again, but then the IP changes and is blocked again.
I would have thought that Google would host their own websites using clean, Google-reserved IPs, but I suppose that is not the case.

We looked into whitelisting the Angular website in the firewall, but our firewall package does not include dynamically whitelisting IPs associated with a given DNS record. And since the IP of the website is not static, manually whitelisting the specific IP is not feasible.

### Enter the URL of the topic with the problem

The whole site

### Describe what you want to experience that would fix the problem

We would like to ask the operator of the site (I assume this is either someone on Google's Angular team or at some other team at Google) to look into this matter and consider moving towards using IP adresses with good reputation to host the website.

### If this problem caused an exception or error, please paste it here

```true

```

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true

```

### Provide any additional information here in as much as detail as you can

If hosting the site with a good IP is not possible, we would have to host our own mirror of the angular.dev website (at least; maybe we would also need to deploy the v17 aio version, too, since there is still some content not available on adev yet).
This, however, is undesired since we would have to keep track of updates ourselves and we would probably have to hijack the angular.dev and v17.angular.io hostnames in our network to point to our local deployment to be able to click on links on Google and StackOverflow, for example.
I have been able to construct a dockerfile for building and running adev locally by looking into the GitHub actions workflows.
In case this is the route we would have to take I would be thankful for any tips and caveats regarding hosting adev locally.

Thanks so much for looking into this!

Raphael

**Comment by imaksp** [comment]
Hi, I just checked current ip reputation out of curiosity & it seems it is classifying it as abusive ip as it could be proxy, but its not present in any of the blacklist.
https://www.ipqualityscore.com/ip-reputation-check/lookup/199.36.158.100

& Although its showing as Fastly network ip, it is missing from this official list:
https://www.fastly.com/documentation/reference/api/utils/public-ip-list/

**Comment by josephperrott** [comment]
We use public firebase hosting, you can actually see the exact publishing process as code within this repository. The IP address that our DNS record points to is static and does not change.


You can see in [firebase's documentation]( https://firebase.google.com/docs/hosting/custom-domain) that this IP address is the IP address used for all custom domain's with firebase hosting. Unfortunately, unless firebase changes the public IP that our DNS record is expected to point to ours will be required to remain as `199.36.158.100`.

I am going to close out this issue as you will need to work with your IT department to alter how the filtering is occurring as it sounds to me as though this filtering will be blocking all firebase hosted services, not only ours.

---

For a little additional detail of the setup, you can see in our DNS record:
```
; <<>> DiG 9.20.0-1-Debian <<>> -t A angular.dev
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 6377
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;angular.dev.			IN	A

;; ANSWER SECTION:
angular.dev.		300	IN	A	199.36.158.100

;; Query time: 152 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Thu Dec 05 16:57:08 UTC 2024
;; MSG SIZE  rcvd: 67
```

We keep a single A records which points to this public IP.

Firebase then used the TXT record found in the same DNS record to perform its mapping to the specific application
```
; <<>> DiG 9.20.0-1-Debian <<>> -t txt angular.dev
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 13609
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;angular.dev.			IN	TXT

;; ANSWER SECTION:
angular.dev.		300	IN	TXT	"google-site-verification=HraXwypfrj6BZpB2BST47hxuGHbrgtNzmu0OnWZrgDQ"
angular.dev.		300	IN	TXT	"hosting-site=angular-dev-site"
angular.dev.		300	IN	TXT	"google-site-verification=m6I5BSe893xrqnuoFfe-u_aDiHRjdbAwf1dZNE4_ztU"

;; Query time: 176 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Thu Dec 05 16:46:34 UTC 2024
;; MSG SIZE  rcvd: 277
```

One could describe the setup that firebase (and as I understand the majority of hosting providers) use as a reverse-proxy.

**Comment by kienhoefr** [comment]
Hi,

that is very unfortunate. I tried to see if other Firebase-hosted websites are reachable from our network and have found that while `chromium.org`'s A record points to the same IPv4 and therefore is not reachable, it also has a AAAA record and is reachable via IPv6.

Considering `www.angular.dev` does have a AAAA record and the webserver behind it redirects to `angular.dev`, was the AAAA on `angular.dev` left out on purpose?

However, after having a look at the DNS records I have found that https://angular-dev-site.web.app does not redirect towards `angular.dev` and has the AAAA record, so at least I am able to visit the site again. This workaround will work for the time being, but having the real domain working would be awesome.

Thank you for your time and your help!
